gcs/src/main/java/com/google/cloud/hadoop/fs/gcs/GoogleHadoopFSInputStream.java

```diff
@@ -17,232 +17,24 @@
 package com.google.cloud.hadoop.fs.gcs;
 
 import com.google.cloud.hadoop.gcsio.FileInfo;
-import com.google.cloud.hadoop.gcsio.GoogleCloudStorageFileSystem;
 import com.google.cloud.hadoop.gcsio.GoogleCloudStorageReadOptions;
-import com.google.common.base.Preconditions;
-import com.google.common.flogger.GoogleLogger;
 import java.io.IOException;
 import java.net.URI;
-import java.nio.ByteBuffer;
-import java.nio.channels.ClosedChannelException;
-import java.nio.channels.SeekableByteChannel;
-import org.apache.hadoop.fs.FSInputStream;
 import org.apache.hadoop.fs.FileSystem;
 
-/** A seekable and positionable FSInputStream that provides read access to a file. */
-class GoogleHadoopFSInputStream extends FSInputStream {
-
-  private static final GoogleLogger logger = GoogleLogger.forEnclosingClass();
-
-  // All store IO access goes through this.
-  private final SeekableByteChannel channel;
-
-  // Path of the file to read.
-  private URI gcsPath;
-
-  // Number of bytes read through this channel.
-  private long totalBytesRead;
-
-  // Statistics tracker provided by the parent GoogleHadoopFileSystemBase for recording
-  // numbers of bytes read.
-  private final FileSystem.Statistics statistics;
-
-  // Used for single-byte reads.
-  private final byte[] singleReadBuf = new byte[1];
-
-  /**
-   * Constructs an instance of GoogleHadoopFSInputStream object.
-   *
-   * @param ghfs Instance of GoogleHadoopFileSystemBase.
-   * @param gcsPath Path of the file to read from.
-   * @param statistics File system statistics object.
-   * @throws IOException if an IO error occurs.
-   */
+public class GoogleHadoopFSInputStream extends InstrumentatedgoogleHadoopFSInputStream {
```

Comment:
To reduce the diff and not lose history, Can we rename GoogleHadoopFSInputStream file to Base and add a separate FSInput stream file ?